### PR TITLE
Fix topological levels

### PR DESCRIPTION
**Pull request by schrockn**
I was working on ripping out in-memory input step injection and noticed that we weren't returning topological levels correctly.

### CODE REVIEW

**Comment by schrockn** [comment]
other refactoring was happening while i was poking around and I just decided keep them in this diff

File: python_modules/dagster/dagster/core/execution_plan/objects.py [hunk 1]
```diff
@@ -310,8 +310,7 @@ def topological_steps(self):
         return [step for step_level in self.topological_step_levels() for step in step_level]
 
     def topological_step_levels(self):
-        return [
-            [self.step_dict[step_key]]
-            for step_key_level in toposort(self.deps)
-            for step_key in step_key_level
-        ]
+        levels = []
```
Comment (mgasner) [inline python_modules/dagster/dagster/core/execution_plan/objects.py]:
```
[
    [self.step_dict[step_key] for step_key in step_key_level]
    for step_key_level
    in toposort(self.deps)
]```?

File: python_modules/dagster/dagster_tests/core_tests/execution_plan_tests/test_execution_plan.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
+from dagster.core.execution import create_execution_plan
+from .test_multiprocessing import define_diamond_pipeline
+
+
+def test_topological_sort():
```
Comment (mgasner) [inline python_modules/dagster/dagster_tests/core_tests/execution_plan_tests/test_execution_plan.py]: 👍

File: python_modules/dagster/dagster/core/execution_plan/objects.py (hunk 1 above)
Comment (schrockn) [reply python_modules/dagster/dagster/core/execution_plan/objects.py]: sure

**Pull request merged**